pygmt/src/subplot.py

```diff
@@ -0,0 +1,222 @@
+        to set aside space on more than one side (e.g. ``clearance=['w1c',
+        's2c']`` would set a clearance of 1 cm on west side and 2 cm on south
+        side). Such space will be left untouched by the main map plotting but
+        can be accessed by modules that plot scales, bars, text, etc.
+    {J}
+    margins : str or list
+        This is margin space that is added between neighboring subplots (i.e.,
+        the interior margins) in addition to the automatic space added for tick
+        marks, annotations, and labels. The margins can be specified as either:
+
+        - a single value (for same margin on all sides). E.g. '5c'.
+        - a pair of values (for setting separate horizontal and vertical
+          margins). E.g. ['5c', '3c'].
+        - a set of four values (for setting separate left, right, bottom, and
+          top margins). E.g. ['1c', '2c', '3c', '4c'].
+
+        The actual gap created is always a sum of the margins for the two
+        opposing sides (e.g., east plus west or south plus north margins)
+        [Default is half the primary annotation font size, giving the full
+        annotation font size as the default gap].
+    {R}
+    sharex : bool or str
+        Set subplot layout for shared x-axes. Use when all subplots in a column
+        share a common *x*-range. If ``sharex=True``, the first (i.e.,
+        **t**\ op) and the last (i.e., **b**\ ottom) rows will have
+        *x*-annotations; use ``sharex='t'`` or ``sharex='b'`` to select only
+        one of those two rows [both]. Append **+l** if annotated *x*-axes
+        should have a label [none]; optionally append the label if it is the
+        same for the entire subplot. Append **+t** to make space for subplot
+        titles for each row; use **+tc** for top row titles only [no subplot
+        titles].
+    sharey : bool or str
+        Set subplot layout for shared y-axes. Use when all subplots in a row
+        share a common *y*-range. If ``sharey=True``, the first (i.e.,
+        **l**\ eft) and the last (i.e., **r**\ ight) columns will have
+        *y*-annotations; use ``sharey='l'`` or ``sharey='r'`` to select only
+        one of those two columns [both]. Append **+l** if annotated *y*-axes
+        will have a label [none]; optionally, append the label if it is the
+        same for the entire subplot. Append **+p** to make all annotations
+        axis-parallel [horizontal]; if not used you may have to set
+        ``clearance`` to secure extra space for long horizontal annotations.
+
+        Notes for sharex/sharey:
+
+        - Labels and titles that depends on which row or column are specified
+          as usual via a subplot's own ``frame`` setting.
+        - Append **+w** to the ``figsize`` or ``subsize`` argument to draw
+          horizontal and vertical lines between interior panels using selected
+          pen [no lines].
+    title : str
+        While individual subplots can have titles (see ``layout`` or
```

Comment:
What's `layout`?

Reply:
Good catch, it's changed to `sharex`/`sharey` now.